.github/workflows/mac-pkg.yml

```diff
@@ -0,0 +1,111 @@
+name: Sign and Upload Mac Installer
+
```

Comment:
Minor: At one point, there was a big TODO made by github re: adding [a top-level `permissions` block](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#permissions).  But it's ugly, complex to figure out, and maybe the defaults are okay?  The Windows workflow doesn't have one either, so :man_shrugging:

Reply:
Added, thanks!